stablehlo/dialect/StablehloOps.td

```diff
@@ -1438,9 +1438,20 @@ def StableHLO_AllToAllOp : StableHLO_Op<"all_to_all",
     I64Attr:$split_dimension,
     I64Attr:$concat_dimension,
     I64Attr:$split_count,
-    I64ElementsAttr:$replica_groups
+    I64ElementsAttr:$replica_groups,
+    OptionalAttr<StableHLO_ChannelHandle>:$channel_handle
   );
   let results = (outs HLO_Tensor);
+
+  // channel_handle is only used for the SPMD partitioner, so we add a
+  // simplified builder method for convenience.
+  let builders = [
```

Comment:
What does the auto-generated builder for an optional value look like? Surprised we need this for a trailing optional value

Reply:
Here's how it looks like (with some manual formatting):

```C++
void AllToAllOp::build(
    ::mlir::OpBuilder &odsBuilder, 
    ::mlir::OperationState &odsState, 
    ::mlir::TypeRange resultType0, 
    ::mlir::ValueRange operand, 
    /*optional*/::mlir::IntegerAttr split_dimension, 
    /*optional*/::mlir::IntegerAttr concat_dimension, 
    /*optional*/::mlir::IntegerAttr split_count, 
    ::mlir::DenseIntElementsAttr replica_groups, 
    /*optional*/::mlir::mhlo::ChannelHandleAttr channel_handle) {
  odsState.addOperands(operand);
  if (split_dimension) {
    odsState.addAttribute(getSplitDimensionAttrName(odsState.name), split_dimension);
  }
  if (concat_dimension) {
    odsState.addAttribute(getConcatDimensionAttrName(odsState.name), concat_dimension);
  }
  if (split_count) {
    odsState.addAttribute(getSplitCountAttrName(odsState.name), split_count);
  }
  odsState.addAttribute(getReplicaGroupsAttrName(odsState.name), replica_groups);
  if (channel_handle) {
    odsState.addAttribute(getChannelHandleAttrName(odsState.name), channel_handle);
  }
  odsState.addTypes(resultType0);
}
```

Reply:
Ah, so optional doesn't have a default None value. LGTM in that case.